cluster-autoscaler/processors/datadog/nodeinfosprovider/template_only_node_info_provider.go

```diff
@@ -151,11 +155,18 @@ func (p *TemplateOnlyNodeInfoProvider) refresh() {
 // GetFullNodeInfoFromBase returns a new NodeInfo object built from provided base TemplateNodeInfo
 // differs from utils.GetNodeInfoFromTemplate() in that it takes a nodeInfo as arg instead of a
 // nodegroup, and doesn't need to call nodeGroup.TemplateNodeInfo() -> we can reuse a cached nodeInfo.
-func GetFullNodeInfoFromBase(nodeGroupId string, baseNodeInfo *schedulerframework.NodeInfo, daemonsets []*appsv1.DaemonSet, predicateChecker simulator.PredicateChecker, ignoredTaints taints.TaintKeySet) (*schedulerframework.NodeInfo, errors.AutoscalerError) {
+func (p *TemplateOnlyNodeInfoProvider) GetFullNodeInfoFromBase(nodeGroupId string, baseNodeInfo *schedulerframework.NodeInfo, daemonsets []*appsv1.DaemonSet, predicateChecker simulator.PredicateChecker, ignoredTaints taints.TaintKeySet) (*schedulerframework.NodeInfo, errors.AutoscalerError) {
 	pods, err := daemonset.GetDaemonSetPodsForNode(baseNodeInfo, daemonsets, predicateChecker)
 	if err != nil {
 		return nil, errors.ToAutoscalerError(errors.InternalError, err)
 	}
+
+	podTpls, err := p.podTemplateProcessor.GetDaemonSetPodsFromPodTemplateForNode(baseNodeInfo, predicateChecker, ignoredTaints)
```

Comment:
you are right, 
I didn't added in the other path, because I saw that you are using the default autoscaler implementation in that case.
But I can definitely add also the function call to the other case.

Reply:
PR updated,
I think I found a nice way to handle it without duplicating to much codes. please let me know if it is ok.